lib/pages/submission_page.dart

```diff
@@ -0,0 +1,234 @@
+  late ImagePicker _picker;
+  late Image? uploadedImg;
+
+  @override
+  void initState() {
+    super.initState();
+    _picker = ImagePicker();
+    uploadedImg = null;
+  }
+
+  void setError(String msg) => setState(() => error = msg);
+
+  uploadImageToFirebase(String category) async {
+    final storageRef = FirebaseStorage.instance.ref();
+
+    final imageRef = storageRef.child("$category/$imageName");
+    try {
+      await imageRef.putData(bytes!);
+      return true;
+    } on FirebaseException {
+      setError("Failed to upload to database. Please try again later.");
+      return false;
+    }
+  }
+
+  Future loadImageIntoMemory() async {
+    final XFile? image = await _picker.pickImage(source: ImageSource.gallery);
+
+    if (image != null) {
+      Uint8List imageData = await image.readAsBytes();
+      uploadedImg = Image.memory(imageData);
+      setState(() {
+        bytes = imageData;
+        imageName = image.path.split('/').last;
+      });
+    }
+  }
+
+  Future<void> addFound() {
+    Item item = Item(
+        id: "",
+        userId: context.read<MyUser?>()!.uid,
+        timeCreated: Timestamp.now(),
+        description: description,
+        itemName: itemName,
+        location: GeoPoint(widget.itemLoc.latitude, widget.itemLoc.longitude),
+        phone: 1,
+        picture: imageName);
+
+    // Calling the collection to add a new user
+    return FirestoreService().addItemToFirestore(item, false);
+  }
+
+  void submitItem() async {
+    if (loading) {
+      setError("Form already submitted");
+    } else if (!formKey.currentState!.validate()) {
+      setError("Please fill in form fields");
+    } else if (bytes == null) {
+      setError("Invalid or no image uploaded");
+    } else {
+      setState(() {
+        loading = true;
+        error = "";
+      });
+      bool success = await uploadImageToFirebase("$itemName/$imageName");
+      if (success) {
+        await addFound();
+      }
+      routeBack(context);
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final AppTheme theme = Theme.of(context).extension<AppTheme>()!;
+    final BoxDecoration bodyDecoration = BoxDecoration(
+      color: theme.veryLight,
+      borderRadius: const BorderRadius.all(Radius.circular(10)),
+    );
+    itemNameOptions.sort((a, b) => a.compareTo(b));
+
+    Widget buildError() =>
+        Text(error, style: theme.errorStyle, textAlign: TextAlign.center);
+
+    Widget buildHeader() => Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Padding(
+              padding: const EdgeInsets.all(8.0),
+              child: Text("Submit a found item", style: theme.darkTitleStyle),
+            ),
+            IconButton(
+                color: theme.dark,
+                icon: const Icon(Icons.keyboard_return_rounded),
+                onPressed: () => routeBack(context))
+          ],
+        );
+
+    Widget buildForm() => FastForm(
```

Comment:
I would still like to be able to see what image I uploaded. Feels odd to submit without being able to see image selected.